### DISCUSSION TITLE
New User - Coral TPU vs TensorRT GPU

### DISCUSSION THREAD

**Discussion by mrplow** [blakeblackshear/frigate#4482] [General]
Hi All,

I'm planning to use my server (i5-4690K, RAID, Ubuntu 22.04) with Frigate. My server already runs a bunch of docker services, but rarely goes above 5% CPU usage and usually has around 12GB of memory free.

I'm planning on getting three 1080p ip cameras (maybe four). I've already toyed with Frigate by using the "IP Camera" app on my android phone acting as a camera to see how it works. I like it! Took less than 30 min to setup and integrate into home assistant.

I'll mostly be using Frigate to watch my yard for dogs, cats, birds and BEARS, and to trigger automations when someone pulls into my driveway.

My main question is, should I shill out the ~5x MSRP price for a used Coral from ebay? My server has no discrete graphics, and just uses the iGPU. But, with those crazy prices I could get a used Quadro P600 or GeForce GT 1030 for cheaper than a used Coral. These two cards are just the cheap ones I saw in the discussion #3016. If using TensorRT is a good idea, I'm sure I'd need to research more to see which GPU I should go for - hardware decode/encoding and all that other stuff that it needs.

From my reading, it looks as though TensorRT will be integrated into Frigate proper in 0.12 and there are some docker images floating around for testing in the meantime.

Thanks for taking the time to read my post,
Dan

**Comment by NickM-27** [comment]
Of course the TensorRT detection is not implemented in frigate yet so it is difficult to know what the performance may be for it.

If you're looking to buy now then the value proposition of a coral is lessened due to the higher prices, and of course the nvidia GPU will also be able to run encoding/decoding which may be a bonus as well. It would also support other models like compreface face detection.

On the other hand, the GPU will use more considerably more power compared to the highly efficient coral so if power bill is a concern then that could be a big consideration.

**Reply by mrplow** [reply]
Thanks @NickM-27,

Those were my thoughts as well, tough call. I'm not overly concerned with power consumption, so I think I'll go the GPU route since they offer more features overall. I'll start monitoring my local used sites for cheap GPUs.

Regards,
Dan
